pkg/apitype/core.go

```diff
@@ -86,6 +86,18 @@ type DeploymentV2 struct {
 	Manifest ManifestV1 `json:"manifest" yaml:"manifest"`
 	// Resources contains all resources that are currently part of this stack after this deployment has finished.
 	Resources []ResourceV2 `json:"resources,omitempty" yaml:"resources,omitempty"`
+	// InFlightOperations are all operations that were known by the engine to be currently executing.
+	InFlightOperations []OperationV1 `json:"in_flight_operations,omitempty" yaml:"in_flight_operations,omitempty"`
```

Comment:
For valid stacks that aren't in the middle of an update deployment, is it true that we would expect the `InFlightOperations` to always be nil? And that they would only be present during an update, and any time it is not empty, and the update is no longer active, we should consider the checkpoint invalid?

Reply:
This is sort of a tautology, I guess. We can say a stack is "valid" if it has an empty `InFlightOperations`, and then yeah, having a non-empty `InFlightOperations` would render the stack invalid.

In general, it will be possible for stacks outside of an update to have non-empty `InFlightOperations`. I'm not sure we want to commit to call this stack "invalid". The engine will reject it and it'll ask you to do `pulumi stack export | pulumi stack import` to fix it, but there's nothing fundamentally wrong about the stack that would e.g. prevent it from being inspected in meaningful ways by the service.